nexus_constructor/main_window.py

```diff
@@ -153,6 +153,17 @@ def _set_up_tree_view(self):
         self.delete_action.triggered.connect(self.on_delete_item)
         self.delete_action.setEnabled(False)
         self.component_tool_bar.addAction(self.delete_action)
+
+        self.zoom_action = QAction(
+            QIcon(os.path.join("ui", "zoom.svg")),
+            "Zoom To Component",
+            self.component_tree_view_tab,
+        )
+        self.zoom_action.triggered.connect(self.on_zoom_item)
+        self.zoom_action.setEnabled(False)
+        self.component_tool_bar.addSeparator()
+        self.component_tool_bar.addAction(self.zoom_action)
```

Comment:
These changes are highlighting how much copy/pasted code there is in this function. Can we pull it out into an `add_action` method?